lib/seattleflu/db/cli/command/etl/enrollments.py

```diff
@@ -116,57 +117,83 @@ def etl_enrollments(*, action: str):
+                    if swab["type"] == "ClinicSwab":
+                        # Find existing collected samples, or create skeletal 
+                        # sample records containing just the collection barcode
+                        # linked back to this encounter.  Sample manifests 
+                        # generated by the processing lab will usually be 
+                        # loaded later and fill in the rest of the sample record.
+                        LOG.debug(f"Looking up collected sample code «{barcode}»")
+                        identifier = find_identifier(db, barcode)
+
+                        if not identifier:
+                            LOG.warning(f"Skipping collected {swab['type']} sample with unknown barcode «{barcode}»")
+                            continue
+
+                        assert identifier.set_name in expected_collection_identifier_sets, \
+                            f"{swab['type']} sample with unexpected «{identifier.set_name}» barcode «{barcode}»"
+
+                        # XXX TODO: Relationally model sample type after we choose
+                        # a standard vocabulary (LOINC or SNOMED or whatever FHIR
+                        # normalizes?)
+                        #   -trs, 8 May 2019
+                        details = {
+                            "type": swab["type"],
+                        }
+
+                        sample = upsert_sample(db,
+                            collection_identifier = identifier.uuid,
+                            encounter_id          = encounter.id,
+                            additional_details    = details)
+
+                        # XXX TODO: Should this delete existing linked samples which
+                        # weren't mentioned in this enrollment document?  This would
+                        # support the case of an incorrectly included sample from an
+                        # earlier enrollment document being corrected by a later
+                        # enrollment document.
+                        #   -trs, 8 May 2019
+
+                    elif swab["type"] in {"ScannedSelfSwab", "ManualSelfSwab"}:
+                        # Find existing collected kit, or create skeletal kit 
+                        # records containing just the kit barcode linked
+                        # back to this encounter. Self-test manifests generated
+                        # by the processing lab will usually be loaded later 
+                        # and fill in the kit record.
+                        LOG.debug(f"Looking up collected kit code «{barcode}»")
+                        identifier = find_identifier(db, barcode)
+
+                        if not identifier:
+                            LOG.warning(f"Skipping {swab['type']} kit with unknown barcode «{barcode}»")
+                            continue
+                        
+                        if identifier.set_name != 'kits-fluathome.org':
```

Comment:
I suggest making this `if identifier.set_name not in expected_kit_identifier_sets` and defining that new variable up by `expected_collection_identifier_sets`. This will make it easier to find the valid set names for future refactoring to remove the hard-coding (see the comments near the top of this file).